### ISSUE TITLE
Test hangs when in debug mode - VSCode - dotnet 9

### ISSUE THREAD

**Issue by erwinkramer** [thomhurst/TUnit#1246]
If running tests, it works just fine. If i go into debugging instead of running, then it straight up doesn't do anything but hang. 

Might be similar to https://github.com/thomhurst/TUnit/issues/1018, but i got a repo to test with. 

Check out https://github.com/erwinkramer/bank-api/blob/main/BankApi.Tests/Program.cs where i have the 'broken' setup. 

Running in the latest VSCode on Windows with the extensions as written in `extensions.json` and the VSCode setting as specified in `settings.json`.

By the way, it is a dotnet 9 project. 

Trying to cancel the run is also unresponsive when it's hanging:

![image](https://github.com/user-attachments/assets/78ef45c5-fd61-4f6b-ac70-f18e6d00efba)

**Comment by thomhurst** [comment]
Hmm interesting. Do you know if it affects any other IDEs or have you not tried?

**Comment by erwinkramer** [comment]
@thomhurst just tested in `Microsoft Visual Studio Community 2022 (64-bit) - Preview` and that works fine.

**Comment by thomhurst** [comment]
Wonder if it's a VSCode bug then. I'll see if I can replicate - I don't really use VSCode for .NET normally

**Comment by erwinkramer** [comment]
> Wonder if it's a VSCode bug then. I'll see if I can replicate - I don't really use VSCode for .NET normally

Let me know, i guess if you have the same issue i can report back at https://github.com/microsoft/vscode-dotnettools/issues?

**Comment by thomhurst** [comment]
Yeah same happens for me - I reckon it's worth opening an issue with them

**Comment by erwinkramer** [comment]
> Yeah same happens for me - I reckon it's worth opening an issue with them

Done

**Comment by rschili** [comment]
Just came here to say I'm facing the same problem. I started using TUnit on some side projects before using it for something bigger, and my main IDE is vscode. This is almost a deal breaker for me. The only workaround that I found is not using the testing platform and instead build+run/debug the .exe like any application. But that's quite inconvenient.

"Run Test" works fine. "Debug Test" does not even begin executing the test code, so it's not an attach problem, the tests do not run at all.

**Comment by thomhurst** [comment]
There's an issue on the vscode repo. It's been pretty silent though so might be worth making some noise on it:

https://github.com/microsoft/vscode-dotnettools/issues/1616

**Comment by erwinkramer** [comment]
Bump

**Comment by scominoekd** [comment]
I tested the new version (1.17.48) of the C# Dev Kit, and now I think there is an attachment issue.

<img width="1613" alt="Image" src="https://github.com/user-attachments/assets/e500740f-b3c3-4e0d-a657-ac0a4a44184d" />

**Comment by hfroliveira** [comment]
> I tested the new version (1.17.48) of the C# Dev Kit, and now I think there is an attachment issue.
> 
> 

![Image](https://github.com/user-attachments/assets/53346b5e-c595-4656-99bd-b9a1b1b6c583)

With the same version of the C# Dev Kit, I can debug despite the “Failed to attach to process” error.

**Comment by thomhurst** [comment]
Closing this here as it's a VS Code issue 😄
